### PR TITLE
Throw an exception for invalid Picard intervals

### PR DESCRIPTION
**Pull request by ronlevine**
Implements #2008

### CODE REVIEW

**Comment by ronlevine** [comment]
@droazen Please review.

File: src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java [hunk 1]
```diff
@@ -289,17 +289,16 @@ else if ( new File(arg).exists() ) {
             final IntervalList il = IntervalList.fromFile(inputFile);
             isPicardInterval = true;
 
-            int nInvalidIntervals = 0;
             for (final Interval interval : il.getIntervals()) {
-                if ( glParser.isValidGenomeLoc(interval.getContig(), interval.getStart(), interval.getEnd(), true)) {
+                if (interval.getStart() - interval.getEnd() == 1 ) { // remove once a corrected version of the exome interval list is released.
+                    logger.warn("Possible incorrectly converted length 1 interval : " + interval);
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java]: Add unit tests to `IntervalUtilsUnitTest` showing that we throw a `UserException` for most invalid Picard intervals, but don't throw for the `end == start - 1` case.

Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java]: Also create a GATK4 ticket to remove this hack once a corrected version of the exome interval list is published.

File: src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java [hunk 2]
```diff
@@ -289,17 +289,16 @@ else if ( new File(arg).exists() ) {
...
+                }
+                else if ( glParser.isValidGenomeLoc(interval.getContig(), interval.getStart(), interval.getEnd(), true)) {
                     ret.add(glParser.createGenomeLoc(interval.getContig(), interval.getStart(), interval.getEnd(), true));
                 } else {
-                    nInvalidIntervals++;
+                    throw new UserException(inputFile.toString() +  " has an invalid genome location : " + interval) ;
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java]: "genome location" -> "interval"

Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java]: `inputFile.toString()` -> `inputFile.getAbsolutePath()`

**Comment by droazen** [comment]
Review complete, back to @ronlevine

File: src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java (hunk 2 above)
Comment (ronlevine) [inline src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java]: Done.

File: src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java (hunk 1 above)
Comment (ronlevine) [inline src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java]: Done. https://github.com/broadinstitute/gsa-unstable/issues/1438

**Comment by ronlevine** [comment]
@droazen Change made. Back to you.

**Comment by droazen** [comment]
Pushed some changes here and squashed -- will merge once tests pass.

**Pull request merged**